cmake/common/osconfig.cmake

```diff
@@ -0,0 +1,21 @@
+# CMake operating system bootstrap module
+
+include_guard(GLOBAL)
+
+# Set minimum CMake version specific to host operating system, add OS-specific module directory to default search paths,
+# and set helper variables for OS detection in other CMake list files.
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+  set(CMAKE_C_EXTENSIONS FALSE)
+  set(CMAKE_CXX_EXTENSIONS FALSE)
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/windows")
+  set(OS_WINDOWS TRUE)
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+  set(CMAKE_C_EXTENSIONS FALSE)
+  set(CMAKE_CXX_EXTENSIONS FALSE)
```

Comment:
As far as I know, we didn't have any issues with disabling C++ extensions on Linux. It's only C extensions that we have an issue with, as previously discussed with the `darray` tests. We should be able to globally disable C++ extensions, unless there's some other issue with that.

Reply:
Got it - in that case I'd just add it to the Linux branch below as it makes more sense to keep that setting in a single file and not distribute it further.